### PR TITLE
added useAuthentication hook

### PR DESCRIPTION
**Pull request by abelpz**

### CODE REVIEW

File: src/hooks/calls/user/useAuthentication.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const {
+    data: token,
+    error,
+    mutate: setAuth,
+  } = useSWRImmutable("fetchAuth", fetchAuth, options);
+
+  //Fetch function to get user data
+  const fetchUser = () => {
+    if (!username) throw new Error("missing username");
+    const user = _userClient.userGetCurrent().then(({ data }) => data);
```
Comment (mandolyte) [inline src/hooks/calls/user/useAuthentication.js:44]: Does an "await" need to be here?

File: src/hooks/calls/user/useAuthentication.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  useEffect(() => {
+    if (username) setUser()
+  },[username,setUser]);
+
+  useEffect(() => {
+    if (username && password && tokenName) {
+      setAuth()
+    }
+  },[username,password,tokenName,setUser,setAuth])
```
Comment (mandolyte) [inline src/hooks/calls/user/useAuthentication.js:66]: setUser is in the dependency list but is not used inside the useEffect.

File: src/hooks/calls/user/useAuthentication.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if (!username || !password || !tokenName) throw new Error("missing tokenName, username or password");
+
+    const allTokens = await _userClient
+      .userGetTokens(username)
+      .then(({ data }) => data);
+    const appTokens = allTokens?.filter((item) => item.name === tokenName);
+
+    if (appTokens?.length > 0) {
+      appTokens.forEach((token) => {
+        _userClient
```
Comment (mandolyte) [inline src/hooks/calls/user/useAuthentication.js:20]: Is an "await" needed here?

File: src/hooks/calls/user/useAuthentication.js (hunk 1 above)
Comment (abelpz) [reply src/hooks/calls/user/useAuthentication.js:44]: Actually, SWR does the awaiting in this case. (https://swr.vercel.app/docs/data-fetching), that's why useAuthentication is returning data and not a promise.

File: src/hooks/calls/user/useAuthentication.js (hunk 2 above)
Comment (abelpz) [reply src/hooks/calls/user/useAuthentication.js:66]: You are right, it was used in there and I then moved it to a different useEffect. Thanks!

File: src/hooks/calls/user/useAuthentication.js (hunk 3 above)
Comment (abelpz) [reply src/hooks/calls/user/useAuthentication.js:20]: That script is for deleting old tokens, we are actually not using the response from those requests, and if that request fails it won't brake anything, so I think we shouldn't wait for that to finish.

File: src/hooks/calls/user/useAuthentication.js (hunk 1 above)
Comment (abelpz) [reply src/hooks/calls/user/useAuthentication.js:44]: If you return a Promise SWR will wait for that and return the data.

**Review by mandolyte — APPROVED** [review]

**Pull request merged**